### ISSUE TITLE
building with internal routing support fails on musl distro like Alpine

### ISSUE THREAD

**Issue by andrekeller** [unbit/uwsgi#1220]
```
apk --no-cache add python3 mailcap pcre python3-dev build-base linux-headers pcre-dev
python3 -m ensurepip --default-pip
pip3 install -e git+git://github.com/unbit/uwsgi.git#egg=uwsgi
```

```
    plugins/router_basicauth/router_basicauth.c: In function 'htpasswd_check':
    plugins/router_basicauth/router_basicauth.c:73:5: error: 'struct crypt_data' has no member named 'current_salt'
       cd.current_salt[0] = ~cpwd[0];
         ^
```

**Comment by xrmx** [comment]
This should do the trick
```
diff --git a/plugins/router_basicauth/router_basicauth.c b/plugins/router_basicauth/router_basicauth.c
index 8306fbe..6118b16 100644
--- a/plugins/router_basicauth/router_basicauth.c
+++ b/plugins/router_basicauth/router_basicauth.c
@@ -66,7 +66,7 @@ static uint16_t htpasswd_check(char *filename, char *auth) {
 
                if (clen > 13) cpwd[13] = 0;
 
-#if defined(__linux__) && !defined(__UCLIBC__)
+#if defined(__GLIBC__)
                struct crypt_data cd;
                cd.initialized = 0;
                // we do as nginx here
```

**Comment by andrekeller** [comment]
No really, next error:

```
plugins/router_basicauth/router_basicauth.c: In function 'htpasswd_check':
plugins/router_basicauth/router_basicauth.c:76:46: error: 'ur_basicauth_crypt_mutex' undeclared (first use in this function)
   if (uwsgi.threads > 1) pthread_mutex_lock(&ur_basicauth_crypt_mutex);
                                              ^
plugins/router_basicauth/router_basicauth.c:76:46: note: each undeclared identifier is reported only once for each function it appears in

```

**Comment by xrmx** [comment]
Yeah, need some more grepping for linux but not uclibc pattern then :)

**Comment by xrmx** [comment]
Please give it a try to git://github.com/xrmx/uwsgi.git@moremusl#egg=uwsgi

**Comment by andrekeller** [comment]
Seems to work for me now, thanks!

**Comment by xrmx** [comment]
Fixes pushed to both master and uwsgi-2.0.

**Comment by grooverdan** [comment]
FYI upstream nginx just removed the code:

https://trac.nginx.org/nginx/changeset/eca3e054e978c2a38f081f81edf8086e99d1e7b0/nginx
